### PR TITLE
Fix SDL doesn't send EndService for Video/Audio Service upon IGNITION_OFF #1409

### LINKED ISSUE #1409: [VS][AS] SDL doesn't send EndService for Video/Audio Service upon IGNITION_OFF
**Occurrence:** Always
**Preconditions:**
1. SDL and HMI are started
2. SPT is registered using v3 protocol and activated on HMI

**Steps:**
1. Start Video Service -> SDL send StartServiceACK to mobile
2. Start Audio service -> SDL send StartServiceACK to mobile
3. Send IGNITION_OFF from HMI: On HTML5 HMI click Exit Application and select Ignition Off from drop down

**Expected result:**
SDL sends End Service (Control Frame 0x04) for: RPC, Audio and Video services

**Actual result:**
SDL sends End Service for RPCservice only

**Environment:**
ATF: aa5beae6ab1726a9b43ed2fa15f66c20d7f7883f
SDL: 7bd5326e8a6f5127220bf938a5be3063bdbb4ab4

### PR DESCRIPTION
**Pull request by Ypostolov**
Fixes #1409 

This PR is ready for review.

### Risk
This PR makes no API changes.

### Testing Plan
Changes in PR are already covered within existing unit tests.

### Summary
Fixed SDL doesn't send EndService for Video/Audio Service upon IGNITION_OFF

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/src/application_manager_impl.cc [hunk 1]
```diff
@@ -2545,6 +2545,8 @@ void ApplicationManagerImpl::UnregisterApplication(
     MessageHelper::SendUnsubscribedWayPoints(*this);
   }
 
+  EndNaviServices(app_id);
```
Comment (AByzhynar) [inline src/components/application_manager/src/application_manager_impl.cc]: @Ypostolov What if application did not use AUDIO/VIDEO services? 
 (f.e. application can use RPC service ONLY)

Comment (Ypostolov) [reply src/components/application_manager/src/application_manager_impl.cc]: @AByzhynar 
In this case (application did not use AUDIO/VIDEO services) this function do nothing, just print error message in log : "No info about navi servicies for app".

File: src/components/protocol_handler/src/protocol_handler_impl.cc [hunk 2]
```diff
@@ -596,6 +596,12 @@ void ProtocolHandlerImpl::SendEndServicePrivate(int32_t primary_connection_id,
                                                 uint8_t session_id,
                                                 uint8_t service_type) {
   LOG4CXX_AUTO_TRACE(logger_);
+  if (0 == connection_id) {
+    LOG4CXX_WARN(logger_,
```
Comment (ValeriiMalkov) [inline src/components/protocol_handler/src/protocol_handler_impl.cc]: @Ypostolov Please make constant variable for more understanding same as in this [pull request](https://github.ford.com/SYNC/sdl_core/pull/357/files)

File: src/components/protocol_handler/src/protocol_handler_impl.cc [hunk 3]
```diff
@@ -596,6 +596,12 @@ void ProtocolHandlerImpl::SendEndServicePrivate(int32_t primary_connection_id,
...
+                 "Secondary connection id is unknown. Will be used"
+                 " primary_connection_id");
+    connection_id = primary_connection_id;
+  }
```
Comment (jacobkeeler) [inline src/components/protocol_handler/src/protocol_handler_impl.cc]: Remove this, the correct fix for this is in #2607

**Comment by vkushnirenko-luxoft** [comment]
Rebuild required

**Comment by zhouxin627** [comment]
@Ypostolov 
Hi, there is the same issue with my project and I'm sure that this change can help me to resolve it. 
Do you have any plan to merge this change into the release branch？

**Comment by Ypostolov** [comment]
@zhouxin627 
Hi!
Unfortunately, I do not have this kind of information.

**Comment by zhouxin627** [comment]
@theresalech @jacobkeeler Hi, there is the same issue with my project and I'm sure that this change can help me to resolve it. Do you have any plan to merge this change into the release branch？

**Comment by jacobkeeler** [comment]
@zhouxin627 Just manually removed the unrelated changes from the PR and merged with develop to bring it up to date. Can you verify if the issue is still fixed by this PR?

**Comment by zhouxin627** [comment]
@jacobkeeler  Thank you for your response. I think the issue is still fixed by this PR.

**Pull request merged**